src/components/AqiGraph/AqiGraph.tsx

```diff
@@ -42,12 +43,53 @@ const AqiGraph: ({sensorDocId}: GraphProps) => JSX.Element = ({
   const [yAxisLimit, setYAxisLimit] = useState(defaultYLimit);
   const [yAxisTicks, setYAxisTicks] = useState<number[]>([]);
   const [horizontalFill, setHorizontalFill] = useState<string[]>([]);
+  const [lastValidDate, setLastValidDate] = useState('');
+  const [lastValidTime, setLastValidTime] = useState('');
   const {currentColorScheme} = useColor();
   const {t} = useTranslation(['graph', 'aqiTable']);
 
+  /**
+   * @param year - a number, the year of the date to be formatted
+   * @param month - a number between 1 and 12 (inclusive)
+   * @param day - a number between 1 and 31 (inclusive)
+   * @returns a string for the date written as 'month/day/year' (ex 4/7/2021)
+   */
+  function formatDate(year: number, month: number, day: number) {
+    const formattedDate: string = month + '/' + day + '/' + year;
+    return formattedDate;
+  }
+
+  /**
+   * @param hour - a number between 0 and 24 (inclusive)
+   * @param minutes - a number between 1 and 60 (inclusive)
+   * @returns a string for the time written as 'hour:minute AM/PM' (ex: '12:42 AM')
+   */
+  function formatTime(hour: number, minutes: number) {
+    const hoursPerPeriod = 12;
+
+    let period = ' AM';
+    let leadingZero = '';
+    if (hour === hoursPerDay) {
+      period = 'AM';
+      hour = hoursPerPeriod;
+    } else if (hour >= hoursPerPeriod) {
+      period = ' PM';
+      hour = hour % hoursPerPeriod;
+    }
+    /* eslint-disable-next-line no-magic-numbers */
+    if (minutes < 10) {
+      leadingZero = '0';
+    }
+    if (hour === 0) {
+      hour = hoursPerPeriod;
+    }
+    const formattedTime: string = hour + ':' + leadingZero + minutes + period;
+    return formattedTime;
+  }
+
   useEffect(() => {
     // Get last 24 hours AQI buffer from sensor doc
-    if (sensorDocId) {
+    if (sensorDocId && isValid) {
```

Comment:
I think we should still show the graph for the last 24 hours even if the sensor is not currently valid

Reply:
Should I move the stuff that I have in the graph UI component to the dial UI component instead?

Reply:
I say yes